test/plugin.test.js

```diff
@@ -0,0 +1,464 @@
+
+afterEach(async () => {
+  try {
+    await this.plugin.disconnect()
+  } catch (e) {
+    throw e
+  }
+})
+
+describe('ilp-plugin-ethereum-asym-server tests', () => {
+  /** 
+   * _estimateFee
+   * 
+   * fee estimate is less than 1.2M gwei
+   *   the 1.2M estimate comes from an upper bound estimate:
+   *
+   *   cost of opening a paychan which is the most expensive tx according to
+   *   estimates on machinomy: 120000 gas
+   *
+   *   cost of gas: 10 gwei
+   */
+  describe('_estimateFee', () => {
+    test('fee estimate is less than 1.2M gwei', async () => {
+      const feeTypes = ['open', 'deposit', 'claim']
+      try {
+        const fees = await Promise.all(feeTypes.map(async (feeType) => {
+          const fee = await this.plugin._estimateFee(feeType)
+          expect(fee.toNumber()).toBeLessThan(1200000) 
+        }))
+      } catch (e) {
+        throw e
+      }
+    })
+  })
+
+  /**
+   * _getAccount
+   *
+   * new account is created in account map 
+   * proxy persists to store on set 
+   */
+  describe('_getAccount', () => {
+    test('new account is created in account map', () => {
+      expect(this.plugin._accounts.get(this.mockAccountName)).toBe(this.mockAccount)
+    })
+
+    test('proxy persists to store on set', async () => {
+      this.mockAccount.isBlocked = true
+      this.plugin._store.unload(this.mockAccountName)
+      expect(this.plugin._store.get(this.mockAccountName)).toBeUndefined()
+      try {
+        await this.plugin._store.close()
+        await this.plugin._store.load(this.mockAccountName)
+      } catch (e) {
+        throw e
+      }
+      expect(this.plugin._store.get(this.mockAccountName)).toBe(JSON.stringify(this.mockAccount))
+    })
+  })
+
+  /**
+   * _preConnect
+   *
+   * does not hang on starting watcher 
+   */   
+  describe('_preConnect', () => {
+    test('does not hang on starting watcher', async () => {
+      try {
+        expect(await this.plugin._preConnect()).toBeUndefined()
+      } catch (e) {
+        throw e
+      }
+    }) 
+  })
+
+  /* _connect
+   * 
+   * account correctly loaded from store
+   * throw on blocked account
+   * ethereum address retrieved from client through 'info' protocol
+   * ethereum address is valid
+   */
+  describe('_connect', () => {
+    test('account correctly loaded from store', async () => {
+      this.mockAccount.isBlocked = true
+      this.plugin._store.unload(this.mockAccountName)
+      expect(this.plugin._store.get(this.mockAccountName)).toBeUndefined()
+      try {
+        await this.plugin._store.close()
+        await this.plugin._connect(this.mockAccountAddress, {})
+      } catch (e) {
+      }
+      expect(this.plugin._accounts.get(this.mockAccountName)).toEqual(this.mockAccount)
+    })
+
+    test('throw on blocked account', async () => {
+      this.mockAccount.isBlocked = true
+      try {
+        await this.plugin._connect(this.mockAccountAddress, {})
+        throw Error('Test Error: should trigger catch')
```

Comment:
Instead of doing this, you can do `expect(this.plugin._connect(this.mockAccountAddress, {})).to.be.rejectedWith('...')`